rhel/parser.go

```diff
@@ -24,22 +26,32 @@ func (u *Updater) Parse(ctx context.Context, r io.ReadCloser) ([]*claircore.Vuln
 		return nil, fmt.Errorf("rhel: unable to decode OVAL document: %w", err)
 	}
 	log.Debug().Msg("xml decoded")
-	protoVuln := func(def oval.Definition) (*claircore.Vulnerability, error) {
-		return &claircore.Vulnerability{
-			Updater:            u.Name(),
-			Name:               def.Title,
-			Description:        def.Description,
-			Issued:             def.Advisory.Issued.Date,
-			Links:              ovalutil.Links(def),
-			Severity:           def.Advisory.Severity,
-			NormalizedSeverity: NormalizeSeverity(def.Advisory.Severity),
-			// each updater is configured to parse a rhel release
-			// specific xml database. we'll use the updater's release
-			// to map the parsed vulnerabilities
-			Dist: releaseToDist(u.release),
-		}, nil
+	protoVulns := func(def oval.Definition) ([]*claircore.Vulnerability, error) {
+		vs := []*claircore.Vulnerability{}
+		for _, affected := range def.Advisory.AffectedCPEList {
+			wfn, err := cpe.Unbind(affected)
+			if err != nil {
+				return nil, err
+			}
+			v := &claircore.Vulnerability{
+				Updater:            u.Name(),
+				Name:               def.Title,
+				Description:        def.Description,
+				Issued:             def.Advisory.Issued.Date,
+				Links:              ovalutil.Links(def),
+				Severity:           def.Advisory.Severity,
+				NormalizedSeverity: NormalizeSeverity(def.Advisory.Severity),
+				Repo:               &claircore.Repository{Name: affected, CPE: wfn},
```

Comment:
@hdonnay I noticed that the value of CPE property is not present in a vulnerability report.  Is CPE property properly handled in database backend?

Reply:
```

"11248": {
      "id": "11248",
      "updater": "rhel-8-updater",
      "name": "RHSA-2020:1372: kernel security and bug fix update (Moderate)",
      "description": "The ...",
      "issued": "2020-04-07T02:00:00+02:00",
      "links": "https://access.redhat.com/errata/RHSA-2020:1372 https://access.redhat.com/security/cve/CVE
-2019-15030 https://access.redhat.com/security/cve/CVE-2019-15031 https://access.redhat.com/security/cve/C
VE-2019-18660 https://access.redhat.com/security/cve/CVE-2019-19527",
      "severity": "Moderate",
      "normalized_severity": "Medium",
      "package": {
        "id": "",
        "name": "kernel-headers",
        "version": "",
        "normalized_version": "",
        "cpe": ""
      },
      "distribution": {
        "id": "",
        "did": "rhel",
        "name": "Red Hat Enterprise Linux Server",
        "version": "8",
        "version_code_name": "",
        "version_id": "8",
        "arch": "",
        "cpe": "cpe:2.3:o:redhat:enterprise_linux:8:*:*:*:*:*:*:*",
        "pretty_name": "Red Hat Enterprise Linux Server 8"
      },
      "repository": {
        "name": "cpe:/o:redhat:enterprise_linux:8::baseos",
        "cpe": ""
      },
      "fixed_in_version": "0:4.18.0-147.8.1.el8_1"
    },
```